browser/brave_browser_main_parts_mac.mm

```diff
@@ -7,18 +7,28 @@
 
 #include "brave/browser/mac/keystone_glue.h"
 #include "brave/browser/sparkle_buildflags.h"
+#include "brave/browser/updater/buildflags.h"
 
 #if BUILDFLAG(ENABLE_SPARKLE)
 #import "brave/browser/mac/sparkle_glue.h"
 #endif
 
+#if BUILDFLAG(ENABLE_OMAHA4)
+#include "brave/browser/updater/features.h"
+#endif
+
 void BraveBrowserMainPartsMac::PreCreateMainMessageLoop() {
   ChromeBrowserMainPartsMac::PreCreateMainMessageLoop();
 
 #if BUILDFLAG(ENABLE_SPARKLE)
-  // It would be no-op if udpate is disabled.
-  [[SparkleGlue sharedSparkleGlue] registerWithSparkle];
+#if BUILDFLAG(ENABLE_OMAHA4)
```

Comment:
It's confusing that omaha is guarded with sparkle.
sparkle is always enabled when using omaha?